### PR TITLE
Enable complexType handling in SegmentProcessFramework

### PR DESCRIPTION
**Pull request by swaminathanmanish**
**Whats in the PR**:
Enabling complexType (ingestion config) handling in SegmentProcessorFramework

**Why its needed**:
For batch/minion based ingestion SegmentProcessorFramework is the building block which currently does not support [complex type](https://docs.pinot.apache.org/v/release-0.9.0/basics/data-import/complex-type) transformation. With this change, all batch ingestion tasks that use SegmentProcessorFramework, will get the ability to transform complexType.

### CODE REVIEW

**Comment by swaminathanmanish** [comment]
@klsince - Could you take a look ?

File: pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java [hunk 1]
```diff
@@ -290,8 +290,7 @@ private List<File> generateSegment(Map<String, GenericRowFileManager> partitionT
           GenericRowFileRecordReader recordReaderForRange = recordReader.getRecordReaderForRange(startRowId, endRowId);
           SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
           driver.init(generatorConfig, new RecordReaderSegmentCreationDataSource(recordReaderForRange),
-              RecordEnricherPipeline.getPassThroughPipeline(),
-              TransformPipeline.getPassThroughPipeline());
+              RecordEnricherPipeline.getPassThroughPipeline(), TransformPipeline.getPassThroughPipeline());
```
Comment (KKcorps) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java:293]: nit: unnecessary formatting

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/CompositeTransformer.java [hunk 2]
```diff
@@ -83,11 +83,10 @@ public class CompositeTransformer implements RecordTransformer {
    */
   public static List<RecordTransformer> getDefaultTransformers(TableConfig tableConfig, Schema schema) {
     return Stream.of(new ExpressionTransformer(tableConfig, schema), new FilterTransformer(tableConfig),
-            new SchemaConformingTransformer(tableConfig, schema),
-            new SchemaConformingTransformerV2(tableConfig, schema), new DataTypeTransformer(tableConfig, schema),
-            new TimeValidationTransformer(tableConfig, schema), new SpecialValueTransformer(schema),
-            new NullValueTransformer(tableConfig, schema), new SanitizationTransformer(schema)).filter(t -> !t.isNoOp())
-        .collect(Collectors.toList());
+        new SchemaConformingTransformer(tableConfig, schema), new SchemaConformingTransformerV2(tableConfig, schema),
```
Comment (KKcorps) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/CompositeTransformer.java:86]: not: unnecessary formatting

**Review by KKcorps — APPROVED** [review]
minor comments

File: pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java (hunk 1 above)
Comment (swaminathanmanish) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java]: Yes did not mean to change this code . It reformatted when I did a "reformat code". Removed this formatting change.

**Review by klsince — APPROVED** [review]
